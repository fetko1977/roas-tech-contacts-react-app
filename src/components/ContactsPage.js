import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Col, Button, Modal, Tabs, Tab } from 'react-bootstrap';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faIdCardAlt } from "@fortawesome/free-solid-svg-icons";

export class ContactsPage extends Component {
    constructor(props) {
        super(props)
        this.state = {
            selectedContact: {},
            showModal: false
        }
        this.handleButtonClick = this.handleButtonClick.bind(this);
        this.handleClose = this.handleClose.bind(this);
    }

    handleButtonClick (id) {
        const selectedContact = this.props.contacts.find(contact => contact.id === id);
        this.setState({ 
            selectedContact,
            showModal: true
        })
    }

    handleClose() {
        this.setState({
            selectedContact: {},
            showModal: false
        })
    }

    render() {
        const { contacts } = this.props;
        const { showModal, selectedContact } = this.state;
        return (
          <div>
            <div className="contacts">
              <div className="container">
                <div className="row">
                  <Col xs={12} className="contacts__title">
                    <h1>Contacts:</h1>
                  </Col>
                  <Col xs={12}  className="contacts__content">
                    {contacts.length > 0 ? (
                      contacts.map(contact => (
                        <Col xs={12} key={contact.id} className="contacts__contact-item float-left mb-3">
                          <Col>
                            <p>
                              <strong>Name:</strong> {contact.name}
                            </p>
                            <p>
                              <strong>Phone:</strong>
                              {contact.phone}
                            </p>
                          </Col>
                          <Col className="d-flex justify-content-end">
                            <Button
                              variant="primary"
                              onClick={() => this.handleButtonClick(contact.id)}
                            >
                              Details
                            </Button>
                          </Col>
                        </Col>
                      ))
                    ) : (
                      <Col>No contacts.</Col>
                    )}
                  </Col>
                </div>
              </div>
            </div>
            <Modal
              show={showModal}
              onHide={this.handleClose}
              className="contact-info right"
            >
            <div className="modal-close">
                <Button onClick={this.handleClose}>x</Button>
            </div>
              <div className="container">
                <div className="row">
                  <Col xs={12} className="contact-info__icon">
                    <FontAwesomeIcon icon={faIdCardAlt} size="3x" />
                  </Col>
                  <Col xs={12} className="contact-info__name">
                    {selectedContact.name}
                  </Col>
                  <Col xs={12} className="contact-info__tags">
                    <span>ADMIN</span>
                    <span>HOST</span>
                    <span>TAG 3</span>
                  </Col>
                  <Col xs={12} className="contact-info__content">
                    <Tabs
                      defaultActiveKey="maininfo"
                      id="uncontrolled-tab-example"
                    >
                      <Tab eventKey="maininfo" title="Main Info">
                        <Col xs={12} className="contact-info__content-main-info">
                          <Col xs={12} sm={6} className="float-left">
                            <p className="label">Phone</p>
                            <p className="value">{selectedContact.phone}</p>
                          </Col>
                          <Col xs={12} sm={6} className="float-left">
                            <p className="label">Ip</p>
                            <p className="value">192.168.0.1</p>
                          </Col>
                        </Col>
                      </Tab>
                      <Tab eventKey="history" title="History">
                        History
                      </Tab>
                      <Tab eventKey="messages" title="Messages" disabled>
                        Messages
                      </Tab>
                    </Tabs>
                  </Col>
                </div>
              </div>
            </Modal>
          </div>
        );
    }
}

const mapStateToProps = state => ({
    contacts: state.contacts
});
 
export default connect(mapStateToProps, undefined)(ContactsPage);