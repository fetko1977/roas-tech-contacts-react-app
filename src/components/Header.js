import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { logout } from '../actions/auth';
import { Navbar, Nav } from 'react-bootstrap';
import Image from 'react-bootstrap/Image'

export const Header = ({logout}) => (
    <header className="header">
        <div className="container">
            <Navbar expand="lg">
                <Navbar.Brand href="/home">
                    <Image src="images/connect.png" thumbnail />
                </Navbar.Brand>
                <Navbar.Toggle aria-controls="basic-navbar-nav" />
                <Navbar.Collapse id="basic-navbar-nav">
                    <Nav className="mr-auto">
                        <Nav.Link href="/home">Home</Nav.Link>
                        <Nav.Link href="/profile">Profile</Nav.Link>
                        <Nav.Link href="/contacts">Contacts</Nav.Link>
                    </Nav>
                    <Nav className="ml-auto">
                        <button onClick={logout} className="nav-link">Logout</button>
                    </Nav>
                </Navbar.Collapse>
            </Navbar>
        </div>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    logout: () => dispatch(logout())
});

export default connect(undefined, mapDispatchToProps)(Header);