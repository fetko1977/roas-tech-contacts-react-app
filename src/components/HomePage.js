import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Col, Card, Button } from 'react-bootstrap';
import { logout } from '../actions/auth';

export class HomePage extends Component { 
    render() {
        let { user, logout } = this.props;
        return (
            <div className="home">
                <div className="container">
                    <div className="row">
                        <Col xs={12}>
                            <h1>Home</h1>
                        </Col>
                        <Col xs={12} sm={8} md={5} lg={3}>
                            <Card className="p-3">
                            <Card.Img variant="top" src="images/female-user-icon.png" />
                            <Card.Body>
                                <Card.Title style={{fontSize: 20}}>{user.username}</Card.Title>
                                <Card.Subtitle className="mb-2 text-muted">Logged in</Card.Subtitle>
                                <Card.Text>
                                Some quick example text to build on the card title and make up the bulk of
                                the card's content.
                                </Card.Text>
                                <Button variant="primary" onClick={logout} >Logout</Button>
                            </Card.Body>
                            </Card>
                        </Col>
                    </div>
                </div>
            </div>
        );
    }
}

const mapStateToProps = state => ({
    user: state.auth.user
});

const mapDispatchToProps = (dispatch) => ({
    logout: () => dispatch(logout())
});

export default connect(mapStateToProps, mapDispatchToProps)(HomePage);