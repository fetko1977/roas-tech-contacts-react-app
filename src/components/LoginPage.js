import React, { Component } from 'react';
import { connect } from 'react-redux';
import { login } from '../actions/auth';
import { Form, Button, FormGroup, FormControl } from "react-bootstrap";

export class LoginPage extends Component {
    constructor(props) {
        super(props)
        this.state = {
          username: '',
          password: '',
          error: false
        };
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit (event) {
        event.preventDefault();
        const { username, password } = this.state;
        const user = { username, password };

        if (user.username === "Admin" && user.password === "test1A") {
            this.setState({ username: '', password: '', error: false})
            localStorage.setItem('user', JSON.stringify(user));
            this.props.login(user);
            this.props.history.push('/profile');
            return;
        }

        return this.setState({ error: true });
    }

    render() {
        let { username, password, error } = this.state;
        return (
          <div className="container">
            <div className="row">
              <h1 className="box-layout__title">Contacts Application</h1>
              <p>Please use the form below to login.</p>
              <div className="Login">
                { error && <div>Credentials are not valid.</div>}
                <Form onSubmit={this.handleSubmit}>
                  <FormGroup controlId="email">
                    <Form.Label>Username</Form.Label>
                    <FormControl
                      autoFocus
                      type="text"
                      value={username}
                      onChange={e =>
                        this.setState({ username: e.target.value, error: false })
                      }
                    />
                  </FormGroup>
                  <FormGroup controlId="password">
                    <Form.Label>Password</Form.Label>
                    <FormControl
                      value={password}
                      onChange={e => this.setState({ password: e.target.value, error: false })}
                      type="password"
                    />
                  </FormGroup>
                  <Button block type="submit">
                    Login
                  </Button>
                </Form>
              </div>
            </div>
          </div>
        );
    }
};

const mapStateToProps = state => ({
  username: state.username,
  password: state.password,
  error: state.erorr
});

const mapDispatchToProps = (dispatch) => ({
    login: (user) => dispatch(login(user))
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
