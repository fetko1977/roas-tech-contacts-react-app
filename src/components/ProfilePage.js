import React from 'react';

const ProfilePage = () => (
    <div className="profile">
        <div className="container">
            <div className="row">
                <div className="col">
                    <h1>This is Profile Page!!!</h1>
                </div>
            </div>
        </div>
    </div>
)
 
export default ProfilePage;