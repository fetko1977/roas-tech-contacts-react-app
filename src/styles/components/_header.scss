.header {
    background: $dark-grey;
    .container {
        .row {
            .navbar-brand {
                img {
                    width: 100%;
                    max-width: 50px;
                }
            }

            .navbar-nav {
                a {
                    color: $white;
                }
            }

            .header__logo-wrapper {
                display: flex;
                justify-content: space-between;
                align-items: center;
                padding: 10px 15px;

                @media (min-width: $desktop-breakpoint) {
                    padding: 0 15px;
                }

                .navbar-toggler {
                    border: none;
                    color: $white;
                }
            }

            .header__nav-wrapper {
                .nav-link {
                    @media (min-width: $desktop-breakpoint) {
                        padding: 0 20px;
                        border-right: 1px solid $off-white;
                    }
                }

                button.nav-link {
                    background: transparent;
                    border: none;
                    color: $white;
                    border-right: none;
                    padding-right: 0;
                    text-align: left;

                    &:hover,
                    &:active {
                        outline: 0
                    }

                    @media (min-width: $desktop-breakpoint) {
                        text-align: center;
                    }
                }

                .navbar-collapse {
                    border-top: 1px solid $off-white;
                    padding: 10px 0;
                    @media (min-width: $desktop-breakpoint) {
                        border-top: none;
                    }
                }
            }
        }
    }
}