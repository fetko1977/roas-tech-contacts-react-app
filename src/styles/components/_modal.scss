.modal.right.fade .modal-dialog {
    right: 0;
    top: 0;
    bottom: 0;
    height: 100%;
    position: absolute;
    margin: 0 auto;
    
}

.modal-close {
    display: flex;
    justify-content: center;
    align-items: center;
    position: absolute;
    left: 10px;
    top: 10px;
    z-index: 10;

    @media (min-width: $desktop-breakpoint) {
        left: -40px;
        top: 10px;
    }
    button {
        background: $blue;
        width: 30px;
        height: 30px;
        color: $dark-grey;
        padding: 0;
        color: $white;
        @include border-radius(50%);

        @media (min-width: $desktop-breakpoint) {
            background: $off-white;
            color: $dark-grey;
        }
    }
}

.modal-content {
    height: 100%;
    @include border-radius(0);

    .contact-info__icon {
        color: $blue;
        display: flex;
        justify-content: center;
        align-items: center;
        padding: 20px 0;
        height: 15vh;
    }

    .contact-info__name {
        text-align: center;
        font-size: 20px;
        font-weight: bold;
        color: $blue;
        padding: 0;
        height: 5vh;
    }

    .contact-info__tags {
        padding: 0 0 50px;
        display: flex;
        justify-content: center;
        align-items: center;
        height: 10vh;

        span {
            padding: 2px 10px;
            border: 1px solid $blue;
            color: $blue;
            font-size: 12px;
            margin: 0 5px;
            @include border-radius(5px);
        }
    }

    .contact-info__content {
        padding: 0;
        background: $off-white;
        height: 70vh;

        .nav-tabs {
            border: none;
            background: $white;
            justify-content: space-evenly;
            border-bottom: 1px solid $dark-grey;
            .nav-item {
                color: $dark-grey;

                &.active {
                    background: $off-white;
                    border: 1px solid $dark-grey;
                    border-bottom: none;
                    font-weight: bold;
                }

                &.disabled {
                    color: lighten($dark-grey, 20%);
                }

                &:hover {
                    font-weight: bold;
                    border: none;
                }
            }
        }

        .tab-content {
            .contact-info__content-main-info {
                padding: 20px;
                p {
                    text-transform: uppercase;
                    font-size: 12;
                    margin: 0;

                    &.label {
                        color: lighten($dark-grey, 30%);
                    }

                    &.value {
                        color: $dark-grey;
                        font-weight: bold;
                    }
                }
            }
        }
    }
}