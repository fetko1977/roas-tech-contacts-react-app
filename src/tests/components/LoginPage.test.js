import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

let login, wrapper;

beforeEach(() => {
    login = jest.fn();
    wrapper = shallow(<LoginPage login={login}/>);
});

test('should render LoginPage correctly', () => {
    expect(wrapper).toMatchSnapshot();
});