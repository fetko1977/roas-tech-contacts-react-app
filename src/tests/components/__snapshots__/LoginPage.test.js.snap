// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="login-wrapper"
    >
      <div
        className="login-wrapper__header"
      >
        <h1
          className="login-wrapper__title"
        >
          Contacts Application
        </h1>
        <p>
          Please use the form below to login.
        </p>
      </div>
      <div
        className="login-wrapper__form-wrapper"
      >
        <Form
          inline={false}
          onSubmit={[Function]}
        >
          <FormGroup
            controlId="email"
          >
            <FormLabel
              column={false}
              srOnly={false}
            >
              Username
            </FormLabel>
            <FormControl
              autoFocus={true}
              onChange={[Function]}
              type="text"
              value=""
            />
          </FormGroup>
          <FormGroup
            controlId="password"
          >
            <FormLabel
              column={false}
              srOnly={false}
            >
              Password
            </FormLabel>
            <FormControl
              onChange={[Function]}
              type="password"
              value=""
            />
          </FormGroup>
          <button
            className="btn--blue"
            type="submit"
          >
            Login
          </button>
        </Form>
      </div>
    </div>
  </div>
</div>
`;
